Mathlib/RingTheory/Smooth/StandardSmooth.lean

```diff
@@ -0,0 +1,208 @@
+/-
+Copyright (c) 2024 Christian Merten. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jung Tao Cheng, Christian Merten, Andrew Yang
+-/
+import Mathlib.Algebra.MvPolynomial.PDeriv
+import Mathlib.LinearAlgebra.Determinant
+import Mathlib.RingTheory.Presentation
+
+/-!
+# Standard smooth algebras
+
+In this file we define standard smooth presentations for algebras. This notion
+is introduced for `SubmersivePresentations`, which is a `Presentation`
+that has less relations than generators. More precisely there exists
+an injective map from `P.relations` to `P.vars`.
+
+Finally, we introduce the notion of standard smooth algebras, which are algebras that
+admit a standard smooth presentation.
+
+## Main definitions
+
+All of these are in the `Algebra` namespace. Let `S` be an `R`-algebra.
+
+- `SubmersivePresentation`: A `Presentation` of `S` as `R`-algebra, equipped with an injective map
+  from `P.relations` to `P.vars`. This map is used to define the differential of a submersive
+  presentation.
```

Comment:
Is this `P.map`? (And there is an inconsistency between `P.rels` and `P.relations` here and below.)

Reply:
Yes, it's `P.map`, I have made it explicit.